### ISSUE TITLE
[wg/webediting] Web Editing Working Group Charter

### ISSUE THREAD

**Issue by siusin** [w3c/strategy#522]
New charter proposal, reviewers please take note.

# Charter Review

[Charter](https://w3c.github.io/charter-drafts/2025/webediting-wg.html)

[diff from charter template](https://services.w3.org/htmldiff?doc1=https://w3c.github.io/charter-drafts/charter-template.html&doc2=https://w3c.github.io/charter-drafts/2025/webediting-wg.html)

Expected end of charter refinement phase: [December 2025](https://lists.w3.org/Archives/Member/w3c-ac-members/2025JulSep/0062.html)

If applicable:

[diff from previous charter](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fwww.w3.org%2F2023%2F09%2Fwebediting-charter-2023.html&doc2=https%3A%2F%2Fw3c.github.io%2Fcharter-drafts%2F2025%2Fwebediting-wg.html)

[chair dashboard](https://www.w3.org/PM/Groups/chairboard.html?gid=wg/webediting)

What kind of charter is this? Check the relevant box / remove irrelevant branches. 

* Existing 
 - [x] Existing WG recharter 

Horizontal Reviews: apply the Github label "Horizontal review requested" to request reviews for accessibility (a11y), internationalization (i18n), privacy, security, and TAG. **Also add a "card" for this issue to the [Strategy Funnel](https://github.com/orgs/w3c/projects/97/views/2).**

Where would charter proponents like to see issues raised? 
https://github.com/w3c/editing/issues/new/

## Anything else we should think about as we review? 
The WG is thinking about putting together a new doc on how users interact with contentEditable and EditContext, probably around the idea of editing hosts and user interaction.

/cc
@johanneswilm

**Comment by siusin** [comment]
@plehegar 
Could you formally [select a Chartering Facilitator](https://www.w3.org/guide/process/charter.html#cr-acceptance)?

**Comment by plehegar** [comment]
[Charter refinement](https://lists.w3.org/Archives/Member/w3c-ac-members/2025JulSep/0062.html)

**Comment by simoneonofri** [comment]
No comments from Security. There are for sure important implications of the related specs, we are happy to review them.

**Comment by ruoxiran** [comment]
No comments from APA.

**Comment by xfq** [comment]
No comments from i18n.

**Comment by dontcallmedom** [comment]
* the adopted drafts in the deliverables section should be pointing to the latest version of the TR documents (they're not in many cases)
* the virtual keyboard spec seems to have received no activity (not even in github) in the past 4 years - is it still something the WG really intends to work on?
* "Other deliverables" contains a mix of tentative normative deliverables and non-normative ones - since the former have a very distinct IPR impact, I suggest using the same model as other charters with tentative deliverables (e.g. https://www.w3.org/2026/01/social-web-wg-charter.html#deliverables) - this may be a pattern worth encoding in the charter template
* tentative deliverables without a draft or even an explainer aren't ideal for AC review - is there any chance the WG could at least write up a short explainer for Caret and Spellchecker?

**Comment by siusin** [comment]
...
> * the virtual keyboard spec seems to have received no activity (not even in github) in the past 4 years - is it still something the WG really intends to work on?
...
> * tentative deliverables without a draft or even an explainer aren't ideal for AC review - is there any chance the WG could at least write up a short explainer for Caret and Spellchecker?

@johanneswilm any thoughts on these two questions?

**Comment by johanneswilm** [comment]
* The virtual keyboard spec is something Microsoft has been especially interested in in the past. 

* Spellcheck API: This was something Gary Kacmarcik and Microsoft were working on.

Any idea what your plans are going forward with this? @roraja @dandclark

* The caret movement spec (including caret movement in bidi text) is something we last discussed at TPAC and @ragoulik (Microsoft) volunteered to maintain it.   Maybe you could provide an explainer?

**Comment by siusin** [comment]
Note: The Web Editing WG is still in its refinement period.

You can find more details at:
  https://github.com/w3c/editing/issues/527.

**Comment by siusin** [comment]
> * The virtual keyboard spec is something Microsoft has been especially interested in in the past.
> * Spellcheck API: This was something Gary Kacmarcik and Microsoft were working on.
> 
> Any idea what your plans are going forward with this? [@roraja](https://github.com/roraja) [@dandclark](https://github.com/dandclark)
> 
> * The caret movement spec (including caret movement in bidi text) is something we last discussed at TPAC and [@ragoulik](https://github.com/ragoulik) (Microsoft) volunteered to maintain it.   Maybe you could provide an explainer?

@ragoulik
Is there a plan to provide an explainer for the caret movement spec?

**Comment by johanneswilm** [comment]
@michael and me have come up with this preliminary proposal of an explainer:

> The W3C Caret Movement Specification establishes a standardized model for caret navigation across complex web-based text editing environments including those leveraging the EditContext API and traditional contenteditable elements. It resolves longstanding browser inconsistencies by defining precise, accessibility-aware algorithms for several critical scenarios:
>
> * navigating bidirectional (bidi) text sequences where visual order diverges from logical DOM order, ensuring arrow keys follow user-perceived directionality across mixed left-to-right and right-to-left scripts;
> * traversing absolutely positioned editable elements that disrupt normal document flow, specifying how the caret should logically progress between visually adjacent regions irrespective of DOM hierarchy or CSS layout;
> * enabling seamless transitions of the caret between discrete editing hosts (e.g., separate contenteditable containers or EditContext instances), with clear rules for focus movement across editable boundaries;
> * defining consistent behavior when moving around non-editable inline elements such as images, SVGs, or form controls, treating them as atomic units that the caret can navigate before and after, but not into, while preserving screen reader compatibility and keyboard operability.
> * specify which markup should be used to achieve "empty elements" being addressable with the caret (e.g. `<div></br></div>` should be reachable while `<div></div>` should be ignored?)
> * standardize the behavior and display of placeholder labels (many inconsistencies atm)

**Comment by michael** [comment]
I wrote up a first WIP proposal for specifying "empty addressable content fields with placeholders" here: https://github.com/w3c/editing/issues/528

We can discuss in tomorrow's meeting.

**Comment by siusin** [comment]
A draft explainer has been added to the charter, and the group will continue working on it (with the intention of merging it with https://github.com/w3c/editing/issues/529 ):
  https://w3c.github.io/charter-drafts/2025/webediting-wg.html#ig-other-deliverables

The group has received commitments to continue work on the Virtual Keyboard spec. For the Spellcheck API, we have added a note to clarify its current status:
[[
The inclusion of this document in the charter is subject to implementation commitments from Working Group members and may be reconsidered in the future.
]]

**Comment by dontcallmedom** [comment]
thanks @siusin ! I have posted a pull request #779 to address:
> "Other deliverables" contains a mix of tentative normative deliverables and non-normative ones - since the former have a very distinct IPR impact, I suggest using the same model as other charters with tentative deliverables (e.g. https://www.w3.org/2026/01/social-web-wg-charter.html#deliverables) - this may be a pattern worth encoding in the charter template

I think only
> the adopted drafts in the deliverables section should be pointing to the latest version of the TR documents (they're not in many cases)

remains to be addressed

**Comment by siusin** [comment]
The links to the TR documents and Technical Coordination Groups have been updated accordingly -> https://github.com/w3c/charter-drafts/commit/7f30331517577f3024f5c58e19655dd6766ae167

Thank you for the review!

**Comment by plehegar** [comment]
[Proposed W3C Charter: Web Editing Working Group (until 2026-04-30)](https://lists.w3.org/Archives/Public/public-new-work/2026Apr/0001.html)

**Comment by plehegar** [comment]
[Proposed changes in the Web Editing WG re-charter (in response to AC review)](https://lists.w3.org/Archives/Member/member-charters-review/2026Jun/0000.html)

**Comment by plehegar** [comment]
See [member-charter-review](https://www.w3.org/Search/Mail/Member/search?subject=Editing&lists=member-charters-review)

**Comment by plehegar** [comment]
[Call for Participation: Web Editing Working Group Charter approved](https://lists.w3.org/Archives/Public/public-new-work/2026Jul/0006.html)
